server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4476,7 +4477,9 @@ private UserVmVO commitUserVm(final boolean isImport, final DataCenter zone, fin
 
             VMTemplateVO templateVO = _templateDao.findById(template.getId());
             if (templateVO == null) {
-                throw new InvalidParameterValueException("Unable to look up template by id " + template.getId());
+                InvalidParameterValueException ipve = new InvalidParameterValueException("Unable to look up template by id " + template.getId());
+                ipve.add(VirtualMachine.class, vm.getUuid());
+                throw ipve;
```

Comment:
this could be optimized to a line

`return new InvalidParameterValueException("Unable to look up template by id " + template.getId()).add(VirtualMachine.class, vm.getUuid())`